src/browser/api/channel.ts

```diff
@@ -181,6 +181,12 @@ export module Channel {
         const { uuid, name, payload: messagePayload, action: channelAction, providerIdentity } = payload;
         const intendedTargetIdentity = { uuid, name };
 
+        const messagingWindow = getEntityIdentity(identity);
+
```

Comment:
This getEntityIdentity function returns ProviderIdentity.  Maybe we need to change the name. It's confusing. 
Otherwise it's ok for me.

Reply:
@aziz512 what do you think? Seems like that function only returns {uuid, name, isExternal}. Do we give it a different return type?

Reply:
Maybe give it its own type? PotentiallyExternalIdentity feels awkward but would describe it.

Reply:
This function is defined outside of this PR right, we should create a line item for this.